dfdaemon/initializer/initializer.go

```diff
@@ -102,32 +101,37 @@ func cleanLocalRepo(options *options.Options) {
 }
 
 // rotateLog truncates the logs file by a certain amount bytes.
-func rotateLog(logFile *os.File, logFilePath string) {
+func rotateLog(logFile *os.File) {
 	logSizeLimit := int64(20 * 1024 * 1024)
 	for {
 		time.Sleep(time.Second * 60)
-		stat, err := os.Stat(logFilePath)
+		stat, err := logFile.Stat()
 		if err != nil {
-			log.Errorf("failed to stat %s: %s", logFilePath, err)
+			// make sure this error could be printed when logFile is broken
+			fmt.Fprintf(os.Stderr, "failed to stat: %s\n", err)
 			continue
 		}
 		// if it exceeds the 20MB limitation
 		if stat.Size() > logSizeLimit {
 			log.SetOutput(ioutil.Discard)
+			// make sure set log output back to logFile when error raised
+			defer log.SetOutput(logFile)
```

Comment:
Since the function `rotateLog` has a loop without `break` or `return`, the for loop is infinite. If the function is infinite without return, I am wondering if the `defer log.SetOutput(logFile)` will be triggered.

Reply:
@allencloud 

Yes, you're right, it won't be triggered cause this function is  infinite, so i removed this loop from this function and make the function return err so that it could be easy to collect err and printed them.